src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAligner.java

```diff
@@ -141,6 +145,47 @@ public void setIntraCtgOptions() {
         setClip3PenaltyOption(5);
     }
 
+    /**
+     * Indicate that we want bwa-mem to infer pair-end inter-size stat from the sequences given.
+     * Only has effect in paired alignment.
+     */
+    public void inferPairEndStats() {
```

Comment:
The next four methods present a confusing and stateful interface.  I'd suggest leaving them out entirely, and simply provide an alternate constructor that takes PE stats for the innies, and leave it at that.

Reply:
Oh, wait, that's not enough, is it.  You also wanted to provide the ability to suppress inference.  Another constructor with a boolean suppressInference argument?

Reply:
The problem is that some of the methods affect the PE stats for all 4 orientations, and some don't, and some depend on the existing state.  I just think it's going to confuse people.  The other bits of state are all independent of each other.

Reply:
I see so you would rather see a single BwaMemPairEndStats object that grows into a 4-length array only when needed... correct?